src/Controller/PanelsController.php

```diff
@@ -84,10 +84,46 @@ public function view($id = null)
     {
         $this->set('sort', $this->request->getCookie('debugKit_sort'));
         $panel = $this->Panels->get($id);
+        if ($panel->title === 'History') {
+            $request = $this->Panels->Requests->get($panel->request_id);
+            $this->set('currentRequest', $request);
+        }
```

Comment:
Couldn't this be a property that the History panel provides as its `data()` results?

Reply:
It will be required to save the request entity separate from the panels. So that way it will show in the `recent` finder.
If its ok, i think I can do that.

Reply:
Hrm. Could the Request be loaded with `contain()` That would allow `panel->request` to always contain the related request for the panel.

Reply:
That's better. Done!